### PR TITLE
Make channel join/leave system message smaller

### PR DESCRIPTION
**Pull request by buoyad**
Styles the system message to look more like a regular message, with some custom styles to set them apart:

image

also special-cases the Leave system message to replace the username with `You`, which we only had for the join before.

r? @keybase/react-hackers cc @cecileboucheron 

EDIT: after taking another pass, we'll keep the larger notice for the current user joining to bring their attention to `Manage channel subscriptions`

image

### CODE REVIEW

**Comment by chrisnojima** [comment]
@malgorithms

**Comment by cecileboucheron** [comment]
@buoyad 👍

File: shared/chat/conversation/messages/wrapper/container.js [hunk 1]
```diff
@@ -75,13 +75,19 @@ const mergeProps = (stateProps, dispatchProps) => {
     (previous.type === 'text' || previous.type === 'deleted')
 
   const oldEnough =
-    previous &&
-    previous.timestamp &&
-    message.timestamp &&
+    !!previous &&
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/container.js]: why add these !!?

File: shared/chat/conversation/messages/wrapper/container.js [hunk 2]
```diff
@@ -75,13 +75,19 @@ const mergeProps = (stateProps, dispatchProps) => {
...
+    !!previous.timestamp &&
+    !!message.timestamp &&
     message.timestamp - previous.timestamp > howLongBetweenTimestampsMs
 
+  // Always show a timestamp if the previous message is a concise joined/left message
+  const previousIsJoinedLeft =
+    !!previous && (previous.type === 'systemJoined' || previous.type === 'systemLeft')
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/container.js]: why here too

File: shared/chat/conversation/messages/wrapper/container.js (hunk 1 above)
Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/container.js]: Flow gives a lot of errors about `All branches are incompatible` with `boolean` if I remove any of these

File: shared/chat/conversation/messages/wrapper/container.js (hunk 2 above)
Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/container.js]: these aren't necessary, but flow complains about them too if I reverse the order of `oldEnough || previousIsJoinedLeft` on L88

File: shared/chat/conversation/messages/wrapper/container.js (hunk 1 above)
Comment (chrisnojima) [reply shared/chat/conversation/messages/wrapper/container.js]: k, likely fixable by doing const oldEnough = !!(...)

Comment (buoyad) [reply shared/chat/conversation/messages/wrapper/container.js]: done!

**Pull request merged**